src/PrivateDns/PrivateDns/VirtualNetworkLinks/UpdateAzurePrivateDnsVirtualNetworkLink.cs

```diff
@@ -0,0 +1,120 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+namespace Microsoft.Azure.Commands.PrivateDns.VirtualNetworkLinks
+{
+    using System.Collections;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.PrivateDns.Models;
+    using Microsoft.Azure.Commands.PrivateDns.Utilities;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using ProjectResources = Microsoft.Azure.Commands.PrivateDns.Properties.Resources;
+
+    /// <summary>
+    /// Updates an existing zone.
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsVirtualNetworkLink", SupportsShouldProcess = true, DefaultParameterSetName = FieldsParameterSetName), OutputType(typeof(PSPrivateDnsVirtualNetworkLink))]
+    public class UpdateAzurePrivateDnsVirtualNetworkLink : PrivateDnsBaseCmdlet
+    {
+        private const string FieldsParameterSetName = "Fields";
+        private const string ObjectParameterSetName = "Object";
+        private const string ResourceParameterSetName = "ResourceId";
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group in which the Private DNS zone/Virtual Network link exists.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The full name of the Private DNS zone (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/privateDnsZones", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string ZoneName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The full name of the virtual network link.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/privateDnsZones/virtualNetworkLinks", "ResourceGroupName", "ZoneName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The virtual network link object to set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PSPrivateDnsVirtualNetworkLink InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones/virtualNetworkLinks")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Boolean that represents if registration is enabled on the virtual network link.")]
+        [ValidateNotNullOrEmpty]
+        public bool IsRegistrationEnabled { get; set; }
```

Comment:
should be "bool?"

Reply:
I was not sure if nullable bool types are supported in powershell cmdlets or not. will change this.